builtin/mainmenu/common.lua

```diff
@@ -93,30 +66,29 @@ function render_serverlist_row(spec, is_favorite)
 		end
 	end
 
-	local grey_out = not is_server_protocol_compat(spec.proto_min, spec.proto_max)
+	local grey_out = not spec.is_compatible
 
-	local details
-	if is_favorite then
-		details = "1,"
-	else
-		details = "0,"
-	end
+	local details = {}
 
 	if spec.ping then
 		local ping = spec.ping * 1000
 		if ping <= 50 then
-			details = details .. "2,"
+			table.insert(details, "1")
```

Comment:
table.insert is about 7 times slower.

Reply:
No, using a rope is the recommended way if you have to do lots of string concatenations. Try LuaJIT if your performance is bad.

Reply:
Update: done

Reply:
I believe monte means slower than `details[#details + 1] = "1"`